go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1374,25 +1374,43 @@ func commonFlows(bridge *bridgeConfiguration) ([]string, error) {
 			fmt.Sprintf("cookie=%s, priority=105, in_port=%s, ip, pkt_mark=%s "+
 				"actions=ct(commit, zone=%d, nat(src=%s), exec(set_field:%s->ct_mark)),output:%s",
 				defaultOpenFlowCookie, ofPortPatch, ovnKubeNodeSNATMark, config.Default.ConntrackZone, physicalIP.IP, ctMarkOVN, ofPortPhys))
+
 		// table 0, packets coming from pods headed externally. Commit connections with ct_mark ctMarkOVN
 		// so that reverse direction goes back to the pods.
 		dftFlows = append(dftFlows,
 			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
 				"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:%s",
 				defaultOpenFlowCookie, ofPortPatch, config.Default.ConntrackZone, ctMarkOVN, ofPortPhys))
 
-		// table 0, packets coming from host Commit connections with ct_mark ctMarkHost
-		// so that reverse direction goes back to the host.
-		dftFlows = append(dftFlows,
-			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
-				"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:%s",
-				defaultOpenFlowCookie, ofPortHost, config.Default.ConntrackZone, ctMarkHost, ofPortPhys))
+		if config.Gateway.Mode == config.GatewayModeShared {
+			// table 0, packets coming from host Commit connections with ct_mark ctMarkHost
+			// so that reverse direction goes back to the host.
+			dftFlows = append(dftFlows,
+				fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
+					"actions=ct(commit, zone=%d, exec(set_field:%s->ct_mark)), output:%s",
+					defaultOpenFlowCookie, ofPortHost, config.Default.ConntrackZone, ctMarkHost, ofPortPhys))
+		} else {
+			// table 0, any packet coming from OVN send to host in LGW mode, host will take care of sending it outside if needed.
+			// exceptions are traffic for egressIP and egressGW features which will hit the priority 100 flow instead of this.
+			dftFlows = append(dftFlows,
+				fmt.Sprintf("cookie=%s, priority=103, in_port=%s, ip, ip_src=%s, "+
+					"actions=ct(table=4,zone=%d,nat)",
+					defaultOpenFlowCookie, ofPortPatch, physicalIP.IP, HostMasqCTZone))
+
+			// table 0, packets coming from host Commit connections with ct_mark ctMarkHost
+			// so that reverse direction goes back to the host. Before sending it out SNAT to nodeIP since
+			// some packets maybe coming with the masqueradeIP in LGW
+			dftFlows = append(dftFlows,
+				fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
+					"actions=ct(commit, zone=%d, nat(src=%s), exec(set_field:%s->ct_mark)), output:%s",
```

Comment:
@trozet said we can do a masquerade IPTABLE rule that will ensure the SNAT automatically to nodeIP! will try that out, then we don't need to modify this flow and can leave it as is.